build/yamls/flow-visibility/flow-visibility.yaml

```diff
@@ -0,0 +1,418 @@
+        type: vertamedia-clickhouse-datasource
+        access: proxy
+        url: http://clickhouse-clickhouse.flow-visibility.svc:8123
+        editable: true
+        jsonData:
+          server: clickhouse-clickhouse.flow-visibility.svc
+          port: 9000
+          username: $CH_USERNAME
+        secureJsonData:
+          password: $CH_PASSWORD
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: grafana
+  name: grafana
+  namespace: flow-visibility
+spec:
+  selector:
+    matchLabels:
+      app: grafana
+  template:
+    metadata:
+      labels:
+        app: grafana
+    spec:
+      securityContext:
+        fsGroup: 472
+        supplementalGroups:
+          - 0
+      containers:
+        - name: grafana
+          image: grafana/grafana:8.3.3
+          imagePullPolicy: IfNotPresent
+          env:
+            - name: GF_INSTALL_PLUGINS
+              value: "https://github.com/antrea-io/antrea/releases/download/v1.6.0/grafana-sankey-plugin.zip;antreaflowvisibility-grafana-sankey-plugin,grafana-clickhouse-datasource"
+            - name: CH_USERNAME
+              valueFrom:
+                secretKeyRef: 
+                  name: clickhouse-secret
+                  key: username
+            - name: CH_PASSWORD
+              valueFrom:
+                secretKeyRef:
+                  name: clickhouse-secret
+                  key: password
+          ports:
+            - containerPort: 3000
+              name: http-grafana
+              protocol: TCP
+          readinessProbe:
+            failureThreshold: 3
+            httpGet:
+              path: /robots.txt
+              port: 3000
+              scheme: HTTP
+            initialDelaySeconds: 10
+            periodSeconds: 30
+            successThreshold: 1
+            timeoutSeconds: 2
+          livenessProbe:
+            failureThreshold: 3
+            initialDelaySeconds: 30
+            periodSeconds: 10
+            successThreshold: 1
+            tcpSocket:
+              port: 3000
+            timeoutSeconds: 1
+          resources:
+            requests:
+              cpu: 250m
+              memory: 750Mi
+          volumeMounts:
+            - mountPath: /data
+              name: grafana-pv
+            - mountPath: /etc/grafana/provisioning/datasources
+              name: grafana-datasources
+      volumes:
+        - name: grafana-pv
+          persistentVolumeClaim:
+            claimName: grafana-pvc
+        - name: grafana-datasources
+          configMap:
+            name: grafana-datasource
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: grafana
+  namespace: flow-visibility
+spec:
+  ports:
+    - port: 3000
+      protocol: TCP
+      targetPort: http-grafana
+  selector:
+    app: grafana
+  sessionAffinity: None
+  type: LoadBalancer
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: clickhouse-mounted-configmap
+  namespace: flow-visibility
+data:
+  create_table.sh: |
```

Comment:
Would you maintain it in a separate file so it is easier to make changes? Like https://github.com/antrea-io/antrea/blob/main/build/yamls/base/conf/antrea-agent.conf.

Reply:
Hi @jianjuns , we came up with two possible options:

1. As you suggested, following the convention of the other manifests, e.g. flow-aggregator: maintaining a base manifest, and using a script [generate-manifest-flow-aggregator.sh](https://github.com/antrea-io/antrea/blob/main/hack/generate-manifest-flow-aggregator.sh) to inject the configmap data into the generated manifest. Users will apply the generated manifest to deploy the system.

2. Create a script to run a list of commands like below, to create all the required configmap from files and then deploy the Services. Users will run the script to deploy the system. 
```shell
kubectl create namespace flow-visibility
kubectl create configmap conf1 --from-file=file1 --namespace=flow-visibility
kubectl create configmap conf2 --from-file=file2 --namespace=flow-visibility
...
kubectl apply -f flow-visibility.yaml
```

Cons of option1: We will need to include all the [dashboard json files](https://github.com/antrea-io/antrea/pull/3191) into configmap, to enable dashboard provisioning, so that users no longer need to manually upload the dashboards to the UI. In that case, as the json files are pretty long, the generated yaml file will also be long, ~4-5k lines. Not sure if that will be a concern. 

Cons of option2: It's different from the convention of other manifests. If we decide to integrate the deployment of flow-aggregator into flow-visibility.yaml, it will require more changes to happen compare to option1. 

Do you have any opinions on that, from how we would like to maintain the project?

Reply:
Sorry, I missed the question. I saw your changed to option 2. I feel it is less convenient for users as they need to clone our repo.

If we go option 1, how large (in bytes) the yamls will be? Why you think that can be an issue?

Reply:
> I feel it is less convenient for users as they need to clone our repo.

I agree. 

> If we go option 1, how large (in bytes) the yamls will be? Why you think that can be an issue?

180k bytes. I actually don't know whether the file size will be an issue or not, as I found antrea.yml has similar size. 

I guess our previous concern is: In the future, users may want to customize Clickhouse storage configuration, e.g. store in memory/disk; start monitor or not; TTL (related contents in PR #3244 ). In that case, if the generated YAML is too long, it might create extra difficulties for users to find the right place to make those configuration changes.

Reply:
Two commands should be fine.

Do we have an idea how long the yaml can be? I feel we should still provide a yaml, and can also provide a script as an alternative.

Reply:
For the current Grafana dashboards, it should be 4900 lines. 

Agree, we'll work on option 1. I don't feel it's very necessary to have a script if we have a stand-alone yaml?

Reply:
We can do script later if that sounds good to you.